### PR TITLE
Add message read receipts

### PR DESCRIPTION
**Pull request by tsmbl**
1. Add api to check for unread messages w/o authN
2. Add api to track last read message per user in created thread

### CODE REVIEW

File: src/internal/wallet/data-service-wallets.ts [hunk 1]
```diff
@@ -175,12 +176,12 @@ export class DataServiceWalletDappAddresses implements WalletDappAddresses {
   }
 }
 
-export class DataServiceWalletDappMessages implements WalletDappMessages {
+export class DataServiceWalletDappMessages implements WalletMessages {
```
Comment (kevindingens) [inline src/internal/wallet/data-service-wallets.ts:179]: also rename class to `DataServiceWalletMessages`?

File: src/internal/wallet/data-service-wallets.ts [hunk 2]
```diff
@@ -194,6 +195,10 @@ export class DataServiceWalletDappMessages implements WalletDappMessages {
       text: this.textSerde.deserialize(new Uint8Array(it.text)),
     }));
   }
+
+  hasUnread(query: FindThreadByOtherMemberQuery): Boolean {
```
Comment (kevindingens) [inline src/internal/wallet/data-service-wallets.ts:199]: for testing?

File: src/internal/messaging/solana-messaging.ts [hunk 3]
```diff
@@ -247,6 +255,14 @@ export class SolanaThread implements Thread {
   get updatedAt() {
     return new Date(this.dialectAccount.dialect.lastMessageTimestamp);
   }
+
+  setLastReadMessageTime(time: Date): Promise<void> {
+    return Promise.resolve(undefined);
+  }
+
+  get hasUnreadMessages(): boolean {
+    return false;
```
Comment (kevindingens) [inline src/internal/messaging/solana-messaging.ts:264]: TODO?

**Review by kevindingens — APPROVED** [review]
Small comments

File: src/internal/messaging/solana-messaging.ts (hunk 3 above)
Comment (tsmbl) [reply src/internal/messaging/solana-messaging.ts:264]: Actually this is unused, removing for now

File: src/internal/wallet/data-service-wallets.ts (hunk 2 above)
Comment (tsmbl) [reply src/internal/wallet/data-service-wallets.ts]: Removing this for now, unused

**Pull request merged**